camera/src/camera_android.cpp

```diff
@@ -0,0 +1,206 @@
+    {
+        dmLogError("Could not get static method 'getCamera'.");
+        Detach(env);
+        return false;
+    }
+
+    jobject tmp1 = env->CallStaticObjectMethod(g_cameraClass, g_getCameraMethodId, dmGraphics::GetNativeAndroidActivity());
+    g_cameraObject = (jobject)env->NewGlobalRef(tmp1);
+    if(!g_cameraObject)
+    {
+        dmLogError("Could not create instance.");
+        Detach(env);
+        return false;
+    }
+
+    // get reference to startPreview() and stopPreview() methods
+    g_startPreviewMethodId = env->GetMethodID(g_cameraClass, "startPreview", "()V");
+    if(!g_startPreviewMethodId)
+    {
+        dmLogError("Could not get startPreview() method.");
+        Detach(env);
+        return false;
+    }
+    g_stopPreviewMethodId = env->GetMethodID(g_cameraClass, "stopPreview", "()V");
+    if(!g_stopPreviewMethodId)
+    {
+        dmLogError("Could not get stopPreview() method.");
+        Detach(env);
+        return false;
+    }
+
+    Detach(env);
+    return true;
+}
+
+void CameraPlatform_StartCapture(dmBuffer::HBuffer* buffer, CameraType type, CaptureQuality quality, CameraInfo& outparams)
+{
+    if (!g_cameraObject)
+    {
+        Camera_QueueMessage(CAMERA_ERROR);
+        return;
+    }
+
+    outparams.m_Width = (uint32_t)CAMERA_WIDTH;
+    outparams.m_Height = (uint32_t)CAMERA_HEIGHT;
+
+    // As default behavior, we want portrait mode
+    if (outparams.m_Width > outparams.m_Height) {
+        uint32_t tmp = outparams.m_Width;
+        outparams.m_Width = outparams.m_Height;
+        outparams.m_Height = tmp;
+    }
+
+    uint32_t size = outparams.m_Width * outparams.m_Height;
+    dmBuffer::StreamDeclaration streams_decl[] = {
+        {dmHashString64("rgb"), dmBuffer::VALUE_TYPE_UINT8, 3}
+    };
+    dmBuffer::Create(size, streams_decl, 1, buffer);
+
+    g_VideoBuffer = *buffer;
+
+    JNIEnv* env = Attach();
+	env->CallVoidMethod(g_cameraObject, g_startPreviewMethodId);
+	Detach(env);
+}
+
+void CameraPlatform_StopCapture()
+{
+    if (!g_cameraObject)
+    {
+        Camera_QueueMessage(CAMERA_ERROR);
+        return;
+    }
+
+    JNIEnv* env = Attach();
+    env->CallVoidMethod(g_cameraObject, g_stopPreviewMethodId);
+    Detach(env);
+}
+
+void CameraPlatform_UpdateCapture()
+{
+	if(g_frameLock)
+	{
+        int width = CAMERA_WIDTH;
+        int height = CAMERA_HEIGHT;
+        int numChannels = 4;
+        uint8_t* out;
+        uint32_t outsize;
+        dmBuffer::GetBytes(g_VideoBuffer, (void**)&out, &outsize);
+
+        uint32_t* data = (uint32_t*)g_data;
+        for( int y = 0; y < height; ++y)
+        {
+            for( int x = 0; x < width; ++x)
+            {
+                // We get the image in landscape mode, so we flip it to portrait mode
+                int index = (width-x-1)*height*3 + (height-y-1)*3;
+
+                // RGB <- ARGB
+                uint32_t argb = data[y*width+x];
+
+                out[index+0] = (argb>>0)&0xFF; 	// R
```

Comment:
Hmm, looking at this it seems the name "argb" is wrong. It should probably be called "abgr" then?